### PR TITLE
Add support for "star" inclusion/exclusion (include all/exclude all)

### LINKED ISSUE #1001: Support wildcard "include EVERYTHING" and "include NOTHING" for Projection clause
Due to planned changes for auto-exclusion of certain pseudo-virtual fields (namely, `$vector`), we need mechanism to allow specifying "include everything", like so:

```
{"projection": {
	"*" : 1
}}
```

and for symmetry, might as well add its counterpart:

```
{"projection": {
	"*" : 0
}}
```

for "include nothing".

In both cases this would affect all fields stored in underlying `"doc_json"` as well as otherwise special key property, `"_id"`.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds support for 2 special query projections:

```
{ "*": 1} // include everything
{"*": 0} // include nothing
```

as per #1001, as convenience shortcuts that ignore special field (_id, $vector, $vectorize) logic regarding default inclusion/exclusion.

Also prepares for bigger fix, #1005.

**Which issue(s) this PR fixes**:
Fixes #1001

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java [hunk 1]
```diff
@@ -588,6 +588,69 @@ public void happyPath() {
 
     @Test
     @Order(3)
+    public void happyPathWithIncludeAll() {
+      String json =
+          """
+                          {
+                            "find": {
+                              "sort" : {"$vector" : [0.15, 0.1, 0.1, 0.35, 0.55]},
+                              "projection" : {"*" : 1},
```
Comment (kathirsvn) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:597]: Can we add a test case with non-zero and non-one value as well? (to make sure it behaves as true)

**Review by kathirsvn — APPROVED** [review]
LGTM

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java (hunk 1 above)
Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:597]: @kathirsvn Sounds good, will add in suitable place.

Comment (tatu-at-datastax) [reply src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java:597]: Added more unit tests to cover non-0/non-1 numbers, booleans.

**Review by vkarpov15 — COMMENTED** [review]
LGTM
